static-assets/js/tinymce-plugins/template/plugin.js

```diff
@@ -23,387 +23,422 @@
-        }
-        if (hasAnyClasses(dom, n, getSelectedContentClasses(editor))) {
-          n.innerHTML = sel;
+      html = html.replace(new RegExp('\\{\\$' + escape(k) + '\\}', 'g'), v);
+    });
+    return html;
+  };
+  const replaceVals = (editor, scope) => {
+    const dom = editor.dom,
+      vl = getTemplateReplaceValues(editor);
+    global$2.each(dom.select('*', scope), (e) => {
+      global$2.each(vl, (v, k) => {
+        if (dom.hasClass(e, k)) {
+          if (isFunction(v)) {
+            v(e);
+          }
         }
       });
-      replaceVals(editor, el);
-      editor.execCommand('mceInsertContent', false, el.innerHTML);
-      editor.addVisual();
-    };
+    });
+  };
+  const insertTemplate = (editor, _ui, html) => {
+    const dom = editor.dom;
+    const sel = editor.selection.getContent();
+    html = replaceTemplateValues(html, getTemplateReplaceValues(editor));
+    let el = dom.create('div', {}, parseAndSerialize(editor, html));
+    const n = dom.select('.mceTmpl', el);
+    if (n && n.length > 0) {
+      el = dom.create('div');
+      el.appendChild(n[0].cloneNode(true));
+    }
+    global$2.each(dom.select('*', el), (n) => {
+      if (hasAnyClasses(dom, n, getCreationDateClasses(editor))) {
+        n.innerHTML = getDateTime(editor, getCdateFormat(editor));
+      }
+      if (hasAnyClasses(dom, n, getModificationDateClasses(editor))) {
+        n.innerHTML = getDateTime(editor, getMdateFormat(editor));
+      }
+      if (hasAnyClasses(dom, n, getSelectedContentClasses(editor))) {
+        n.innerHTML = sel;
+      }
+    });
+    replaceVals(editor, el);
+    editor.execCommand('mceInsertContent', false, el.innerHTML);
+    editor.addVisual();
+  };
 
-    var global = tinymce.util.Tools.resolve('tinymce.Env');
+  var global = tinymce.util.Tools.resolve('tinymce.Env');
 
-    const getPreviewContent = (editor, html) => {
-      var _a;
-      let previewHtml = parseAndSerialize(editor, html);
-      if (html.indexOf('<html>') === -1) {
-        let contentCssEntries = '';
-        const contentStyle = (_a = getContentStyle(editor)) !== null && _a !== void 0 ? _a : '';
-        const cors = shouldUseContentCssCors(editor) ? ' crossorigin="anonymous"' : '';
-        global$2.each(editor.contentCSS, url => {
-          contentCssEntries += '<link type="text/css" rel="stylesheet" href="' + editor.documentBaseURI.toAbsolute(url) + '"' + cors + '>';
+  const getPreviewContent = (editor, html) => {
+    var _a;
+    let previewHtml = parseAndSerialize(editor, html);
+    if (html.indexOf('<html>') === -1) {
+      let contentCssEntries = '';
+      const contentStyle = (_a = getContentStyle(editor)) !== null && _a !== void 0 ? _a : '';
+      const cors = shouldUseContentCssCors(editor) ? ' crossorigin="anonymous"' : '';
+      const templatesCSS = editor.options.get('templates_css') ?? [];
+      global$2.each([...editor.contentCSS, ...templatesCSS], (url) => {
```

Comment:
This would throw if contentCSS is null or undefined.
```suggestion
      global$2.each([...(editor.contentCSS??[]), ...templatesCSS], (url) => {
```

But also, do we still want the contentCSS here?

Reply:
I checked the scenario and when there is no contentCSS set (from editor initialization) its value is the empty array. Also, if you set a value to contentCSS as we do in the forms engine control, it'll apply to the template content too. Do we want to keep that, or to manually set it on the forms engine controll too?